### PR TITLE
If 484 sortability of maintenance due

### PR DESCRIPTION
**Pull request by florian-paul**

### CODE REVIEW

**Comment by mattmikulina** [comment]
Task linked: [IF-484 Maintenance Dashboard: Column "Days till Maintenance" + "Operating Hours till Maintenance" should be sortable](https://app.clickup.com/t/2177764/IF-484)

File: fusionfrontend/src/app/components/content/maintenance-progressbar/maintenance-progressbar.component.html [hunk 1]
```diff
@@ -35,8 +35,12 @@
...
 
-  <div *ngIf="!maintenanceValue" class="container">
+  <div *ngIf="isNaN(maintenanceValue)" class="container">
     <p-progressBar [showValue]="false" [value]="0" class="unknown-bar" class="light-grey-bar"></p-progressBar>
     <p class="progress-bar-label na-label">n.a.</p>
   </div>
+  <div *ngIf="isZero(maintenanceValue)" class="container">
+    <p-progressBar [showValue]="false" [value]="0" class="unknown-bar" class="light-grey-bar"></p-progressBar>
+    <p class="progress-bar-label na-label">{{ maintenanceValue | germanNumber }}</p>
+  </div>
```
Comment (g1glh0rn) [inline fusionfrontend/src/app/components/content/maintenance-progressbar/maintenance-progressbar.component.html:45]: are you sure, that 0 means "unknown"? Could also mean critical?

File: fusionfrontend/src/app/store/factory-asset-details/factory-asset-details.model.ts [hunk 2]
```diff
@@ -29,6 +29,8 @@ export class FactoryAssetDetails extends Asset {
   openAlertPriority: OispAlertPriority;
   statusHoursOneDay?: StatusHoursOneDay;
   protocol: string;
+  daysTillMaintenance: number;
+  operatingHoursTillMaintenance: number;
```
Comment (g1glh0rn) [inline fusionfrontend/src/app/store/factory-asset-details/factory-asset-details.model.ts:33]: should we really include fields of an asset in the model? Not sure whether this is a good idea.

**Review by g1glh0rn — CHANGES_REQUESTED** [review]
working well in my test. provides quite some benefit to the maintenance dashboard.

**Pull request merged**